src/main/java/com/artozersky/HackerNewsAPI/cache/impl/CacheEntityServiceImpl.java

```diff
@@ -0,0 +1,90 @@
+package com.artozersky.HackerNewsAPI.cache.impl;
+
+
+import com.artozersky.HackerNewsAPI.cache.CacheEntityService;
+import com.artozersky.HackerNewsAPI.model.impl.NewsPostModelImpl;
+
+import java.util.List;
+import java.time.LocalDateTime;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CacheEntityServiceImpl implements CacheEntityService {
+
+    private final CacheEntityImpl cacheEntity;
+
+    @Autowired
+    public CacheEntityServiceImpl(@Value("${cache.size:100}") int maxSize) {
+
+        this.cacheEntity = new CacheEntityImpl(maxSize);
+
+    }
+
+    @Override
+    public NewsPostModelImpl getPostFromCacheById(Long postId) {
+
+        return cacheEntity.get(postId);
+    }
+
+    @Override
+    public List<NewsPostModelImpl> getAllPostsFromCache() {
+
+        List<NewsPostModelImpl> cachedPosts = cacheEntity.getAll();
+
+        System.out.println("inside getAllPostsFromCache size : " + cachedPosts.size());
+
+        boolean isStale = cachedPosts.stream().anyMatch(post -> {
+            int currentElapsedTime = (int) java.time.Duration.between(post.getCreatedAt(), LocalDateTime.now()).toHours();
+            return Math.abs(post.getTimeElapsed() - currentElapsedTime) > 1;//more than one hour difference
+        });
+
+        if(isStale) {
+           this.clearCache();
+           System.out.println("Cache invalidated due to stale data");
+           return List.of();
+        }
+        return cachedPosts;
+    }
+
+    @Override
+    public void putPostInCache(NewsPostModelImpl post) {
+
+        cacheEntity.put(post.getPostId(), post);
+        System.out.println("Storing in Cache: key = " + post.getPostId() + ", value = " + post);
```

Comment:
remove print statements if you want to debug shit please do it to a log. because you did this please implement a log to the server. don't implement a log by your senf but take a ready to use one i trust you to choose a good one. and every place you think a log is good please add logging. be careful about it because methods that needs to be fast and performance critical log is not good in them. so think about it. a log should contain day hour server ip server post message.

Reply:
please think of user configuration for the log that you can put in the application properties such as file path and so on.